### PR TITLE
add normalizers

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Issue # 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [ ] I have read the contribution requirements and fulfill them.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/account/src/Application/Serializer/Normalizer/LanguagePrivilegesArrayDenormalizer.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        return LanguagePrivileges::class.'[]' === $type
+            && $this->serializer->supportsDenormalization($data, substr($type, 0, -2), $format, $context);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setSerializer(SerializerInterface $serializer)
```
Comment (piotrkreft) [inline module/account/src/Application/Serializer/Normalizer/LanguagePrivilegesArrayDenormalizer.php:77]: SerializerAwareTrait

File: module/api/src/Application/Serializer/Normalizer/ExceptionNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Ergonode\Api\Application\Mapper\ExceptionMapperInterface;
+use Ergonode\Api\Infrastructure\Normalizer\ExceptionNormalizerInterface;
+use Ergonode\Api\Application\Exception\FormValidationHttpException;
+use Ergonode\Api\Application\Exception\ViolationsHttpException;
+
+class ExceptionNormalizer implements NormalizerInterface
```
Comment (piotrkreft) [inline module/api/src/Application/Serializer/Normalizer/ExceptionNormalizer.php:19]: I think there's already native SF functionality with this approach but might be wrong

File: module/core/src/Application/Serializer/Normalizer/AbstractEntityNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        if (!$object instanceof AbstractEntity) {
+            throw new InvalidArgumentException(sprintf(
+                'The object must be an instance of "%s"',
+                AbstractEntity::class,
+            ));
+        }
+        $context[$this->getContextKey($object)] = true;
+        $context[AbstractNormalizer::IGNORED_ATTRIBUTES][] = 'aggregateRoot';
+
+        return $this->normalizer->normalize($object, $format, $context);
```
Comment (piotrkreft) [inline module/core/src/Application/Serializer/Normalizer/AbstractEntityNormalizer.php:37]: It seems it's more efficiecnt to have object normalized injected directly and have cacheable interface

File: module/workflow/src/Domain/Entity/AbstractWorkflow.php [hunk 4]
```diff
@@ -32,31 +32,31 @@ abstract class AbstractWorkflow extends AbstractAggregateRoot implements Workflo
     /**
      * @JMS\Type("Ergonode\SharedKernel\Domain\Aggregate\WorkflowId")
      */
-    private WorkflowId $id;
+    protected WorkflowId $id;
```
Comment (piotrkreft) [inline module/workflow/src/Domain/Entity/AbstractWorkflow.php:35]: It does not seem to be necessary?

**Review by piotrkreft — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Pull request merged**